pkg/machines/components/create-vm-dialog/createVmDialog.jsx

```diff
@@ -525,6 +524,12 @@ class CreateVmModal extends React.Component {
 
     onValueChanged(key, value) {
         switch (key) {
+        case 'vmName':
+            if (/\s/.test(value))
+                this.setState({ [key]: value.split(" ").join("_") });
```

Comment:
Any reason why you test it? I think that just replacing it without firstly checking if fine.
And also can this commit please have `Fixes #12893`, thanks :)